exchange/bidder.go

```diff
@@ -528,6 +532,19 @@ func (bidder *bidderAdapter) doRequestImpl(ctx context.Context, req *adapters.Re
 	}
 	bidder.me.RecordOverheadTime(metrics.PreBidder, time.Since(pbsRequestStartTime))
 	httpCallStart := time.Now()
+
+	if tmaxAdjustments != nil && tmaxAdjustments.Enabled && tmaxAdjustments.BidderResponseDurationMin > 0 {
+		if deadline, ok := ctx.Deadline(); ok {
+			bidderTmax := deadline.Add(-time.Duration(int(tmaxAdjustments.BidderNetworkLatencyBuffer) * int(time.Millisecond)))
+			if int(bidderTmax.Sub(time.Now())) < int(tmaxAdjustments.BidderResponseDurationMin)*int(time.Millisecond) {
+				return &httpCallInfo{
+					request: req,
+					err:     errTmaxTimeout,
+				}
+			}
+		}
+
+	}
```

Comment:
Can we move the entire `if` statement into its own function and unit test it separately? Also, can we move the `httpCallStart := time.Now()` after this? If we decide not to send the request and return a `errTmaxTimeout` here, we would've made the `time.Now()` call in vain.
```
536     bidder.me.RecordOverheadTime(metrics.PreBidder, time.Since(pbsRequestStartTime))
537 -   httpCallStart := time.Now()
538
539 -   if tmaxAdjustments != nil && tmaxAdjustments.Enabled && tmaxAdjustments.BidderResponseDurationMin > 0 {
540 -       if deadline, ok := ctx.Deadline(); ok {
541 -           bidderTmax := deadline.Add(-time.Duration(int(tmaxAdjustments.BidderNetworkLatencyBuffer) * int(time.Millisecond)))
542 -           if int(bidderTmax.Sub(time.Now())) < int(tmaxAdjustments.BidderResponseDurationMin)*int(time.Millisecond) {
543 -               return &httpCallInfo{
544 -                   request: req,
545 -                   err:     errTmaxTimeout,
546 -               }
547 -           }
548 -       }
549 -
550 -   }
    +   if err := aFunction(ctx, tmaxAdjustments); err != nil {
    +       return &httpCallInfo{
    +           request: req,
    +           err:     err,
    +       }
    +   }
    +   httpCallStart := time.Now()
551     httpResp, err := ctxhttp.Do(ctx, bidder.Client, httpReq)
exchange/bidder.go
```

Reply:
Fixed here - https://github.com/prebid/prebid-server/pull/2787/commits/2fd4e44b69e5ad091859df73d9bc2b7b55c57e31